### ISSUE TITLE
Practical Deep Learning course, Feb 2022

### ISSUE THREAD

**Issue by mvsjober** [csc-training/intro-to-dl#7]
Recurring TODOs:
- [x] Reserve Dogmi (Mats)
- [x] Fill course form (Mats)
- [x] Create course project (Markus)
- [x] Training accounts (Markus)
- [x] GPU reservation (Markus)
- [x] Eventilla page (Marina)
- [x] PRACE page (Marina)
- [x] Notebooks course request (Mats)
- [x] Update/review lectures (Mats)
- [x] Update/review lectures (Markus)
- [x] Make tinyurl (Mats)
- [x] Test Notebooks + exercises
- [x] Test Puhti exercises
- [x] Test with training accounts
- [x] Test day1 with Google colab
- [x] Create Zoom room (Mats)

After registration has closed (recurring):
- [x] Non-haka emails to Notebooks team (Mats)
- [x] Send email to participants with: zoom-link, materials link, note to test notebooks (and ssh client?)

During course:
- [x] Remember to record lectures!!

After course:
- [x] Post lecture recordings
- [x] Send attendance info to Marina (→ certificates)

Specific to this instance:
- [x] Test day1 with CSC Notebooks BETA (Katja)
- [x] Verify that all Keras exercises use functional API
- [x] New and updated Transformers exercises (Markus)
- [x] Update day2 exer. instructions
- [x] Add Tensorboard web UI instructions
- [x] Update lecture 5 (Mats)
- [x] Update lecture 6 (Katja)
- [x] Update day2 exercises (Markus)
- [x] Add PyTorch DDP examples, possibly add slide to lecture 7 (Mats)
- [x] Update lecture 7 slides (Markus)
- [x] Investigate PyTorch Lightning, possibly mention in lecture 5 (Mats)
- [x] Fix Notebooks image name at https://github.com/csc-training/intro-to-dl/blob/master/day1/README.md 
- [x] Test puhti web UI workflow for exercises, will this be the "default" in the course?
- [x] Possibly create PyTorch versions of new scripts? (ViT, BiT)
- [x] Fix Exercise_X (python files missing, etc) 
- [ ] Consider using NVME in all scripts?

**Comment by mvsjober** [comment]
Notes from day 1:
- people keep asking content questions also in Zoom, not sure what to do about that... we could simply refuse to answer non-technical questions in Zoom ;-)
- Lecture 3: took only ~45 mins, and we had almost too much time for exercise. We could change schdule like this:

  * 13:00-13:45 Lecture 3 (15 min less)
  * 13:45-14:15 Exercise 3
  * 14:15-14:30 Break
  * 14:30-15:30 Lecture 4 (15 min more!)
  * 15:30-16:00 Exercise 4
 
- some errors in exercise 2, with references to history instead of history_ex1 etc

**Comment by kmankine** [comment]
Comment from a student: We could spend a minute about how to save and transfer files from Notebooks to somewhere safe, and remind people about the notebook lifetime.

**Comment by mvsjober** [comment]
Re-use, finetuning should be explained better. I should use the image in the lecture better. Also the code should be simplified (remove the non-vgg models etc).

**Comment by mvsjober** [comment]
Take a look at PyTorch vs Keras accuracy differences in CNN/RNN 20ng exercises.

**Comment by jmakoske** [comment]
> Comment from a student: We could spend a minute about how to save and transfer files from Notebooks to somewhere safe, and remind people about the notebook lifetime.

I guess this should be explicitly somewhere written down. I mean we were supposed to talk about this but (again) did not.

**Comment by jmakoske** [comment]
> Notes from day 1:
> 
>     * people keep asking content questions also in Zoom, not sure what to do about that... we could simply refuse to answer non-technical questions in Zoom ;-)
> 
>     * Lecture 3: took only ~45 mins, and we had almost too much time for exercise. We could change schdule like this:
>       
>       * 13:00-13:45 Lecture 3 (15 min less)
>       * 13:45-14:15 Exercise 3
>       * 14:15-14:30 Break
>       * 14:30-15:30 Lecture 4 (15 min more!)
>       * 15:30-16:00 Exercise 4
> 
>     * some errors in exercise 2, with references to history instead of history_ex1 etc

* L1 was too long again. Maybe the SGD stuff should be moved to L2?
* Zoom chat may be easier for some? I personally don't mind answering in both chats.

**Comment by jmakoske** [comment]
Move any new git commits to mls

**Comment by mvsjober** [comment]
Moved comments as TODOs to new ticket, closing this one.
